R/clean_redcap.R

```diff
@@ -157,18 +149,34 @@ distill_nonrepeat_table <- function(form_name,
     my_fields <- c(my_fields, "redcap_survey_identifier")
   }
 
-  if (has_repeating) {
+  if (has_repeat_forms) {
     db_data <- db_data %>%
       filter(is.na(.data$redcap_repeat_instance))
   }
 
-  db_data %>%
-    select(all_of(my_fields)) %>%
+  out <- db_data %>%
+    add_partial_keys() %>%
+    select(
+      all_of(my_fields), any_of(c("redcap_event", "redcap_arm", "redcap_form_instance", "redcap_event_instance"))
+    ) %>%
+    relocate(
+      any_of(c("redcap_event", "redcap_arm", "redcap_form_instance", "redcap_event_instance")),
+      .after = all_of(my_record_id)) %>%
```

Comment:
The `relocate` happens because of the creation of new columns via `mutate` in `add_partial_keys` and then also because of the `select` function that happens above this `relocate`. The `rename_with` keeps the order where `redcap_form_instance` exists correctly, but then `select` messes it up.

Reply:
Oh I see. The `select()` reorders anyways and it has to be that way because the record id is in `my_fields`. Makes sense now